sys/include/usb/dfu.h

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (C) 2020 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @defgroup    usb_dfu   DFU - USB Device Firmware Upgrade
+ * @ingroup     usb
+ * @brief       Generic USB DFU defines and helpers
+ *
+ * @{
+ *
+ * @file
+ * @brief       Definition for USB DFU interfaces
+ *
+ * @author      Dylan Laduranty <dylan.laduranty@mesotic.com>
+ */
+
+#ifndef USB_DFU_H
+#define USB_DFU_H
+
+#include <stdint.h>
+
+#include "usb.h"
+#include "usb/descriptor.h"
+#include "usb/usbus.h"
+#include "usb/usbus/control.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name USB DFU RAM information for riotboot
+ * @{
+ */
+#ifndef RIOTBOOT_DFU_ADDR
+#define RIOTBOOT_DFU_ADDR               (CPU_RAM_BASE + CPU_RAM_SIZE - 4)   /**< DFU default magic address */
+#endif
+#ifndef RIOTBOOT_MAGIC_NUMBER
+#define RIOTBOOT_MAGIC_NUMBER           (0xDEADBEEFUL)                      /**< DFU default magic value */
```

Comment:
Don't we have something better for this? This matches the canary used for SSP.

Reply:
Took a random one when I started this PR months ago.
I can switch to whatever you want.